src/utils/web_byte_utils.ts

```diff
@@ -172,8 +172,42 @@ export const webByteUtils = {
     return new TextEncoder().encode(text);
   },
 
-  toUTF8(uint8array: Uint8Array, start: number, end: number): string {
-    return new TextDecoder('utf8', { fatal: false }).decode(uint8array.slice(start, end));
+  toUTF8(uint8array: Uint8Array, start: number, end: number, fatal: boolean): string {
+    if (uint8array.length === 0) {
+      return '';
+    }
+
+    const stringByteLength = end - start;
+    if (stringByteLength === 0) {
+      return '';
+    }
+
+    if (stringByteLength < 200) {
+      let basicLatin = true;
+      const latinBytes = [];
+      for (let i = start; i < end; i++) {
+        const byte = uint8array[i];
+        if (byte > 127) {
+          basicLatin = false;
+          break;
+        }
+        latinBytes.push(byte);
+      }
+
+      if (basicLatin) {
+        // eslint-disable-next-line prefer-spread
+        return String.fromCharCode.apply(String, latinBytes);
```

Comment:
I'd be pretty curious to see how this compares against
```suggestion
        return String.fromCharCode.apply(String, uint8array);
```

or

```suggestion
        return String.fromCharCode.apply(String, Array.from(uint8array));
```

as creating an Array and then repeatedly `.push()`ing into it can also be quite a bit slower than using a pre-generated or fixed-length array

(I can also try myself at some point but you probably have the benchmarking setup working already)

Reply:
I attached benchmarks to the ticket, the take away is that pushing to an array is faster than pre-allocating it or just assigning to the next index of an empty array (`toUTF8_assign_array`). I would guess this has something to do with creating an array with empty slots vs no slots and filling it with a consistent type (always numbers). 

Just guessing though, it is revealed that attaching strings via concat is super fast for small cases, so I've hard-coded 1, 2, and 3 length strings.